test/Microsoft.AspNet.Mvc.DataAnnotations.Test/Internal/DataAnnotationsModelValidatorProviderTest.cs

```diff
@@ -99,13 +100,24 @@ public void IValidatableObjectGetsAValidator()
             var mockValidatable = new Mock<IValidatableObject>();
             var metadata = _metadataProvider.GetMetadataForType(mockValidatable.Object.GetType());
 
-            var providerContext = new ModelValidatorProviderContext(metadata);
+            var providerContext = new ModelValidatorProviderContext(metadata, GetValidatorItems(metadata));
```

Comment:
Is there an scenario in which a user needs to create a ModelValidatorProviderContext?
GetValidatorItems seems like a product of metadata, which could be computed internally on the constructor. I know we want to keep context classes as simple as possible, but doing it in the constructor reduces the complexity of the context class and helps prevents incorrect usages, we can keep the two parameter constructor as an override of the default logic if it makes sense or helps in other areas like unit testing, bear in mind that I don't have a lot of context on this area. @rynowak thoughts?

I also have seen the lines in GetValidatorItems inside the ValidatorCache (I think code) so seems to me that this is logic we care about/might be common in this area